File: demo-vk-rest-scraping.java/src/main/java/demo/model/Faculty.java
package demo.model;

import java.util.List;

public class Faculty {

    private final int id;
    private final String title;

    private Integer chairsCount;
    private List<Chair> chairs;

    private Faculty(int id, String title) {
        this.id = id;
        this.title = title;
    }

    public Faculty(ConstructorMediator cm) {
        this(cm.getId(), cm.getTitle());
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public Integer getChairsCount() {
        return chairsCount;
    }

    public List<Chair> getChairs() {
        return chairs;
    }

    public void setChairs(List<Chair> chairs) {
        this.chairs = chairs;
        this.chairsCount = chairs.size();
    }

    @Override
    public String toString() {
        return "Faculty{" +
                "id='" + id + '\'' +
                ", title='" + title + '\'' +
                '}';
    }
}

File: demo-vk-rest-scraping.kotlin/src/main/kotlin/demo/model/Faculty.kt
package demo.model

class Faculty private constructor(val id: Int, val title: String) {

    var chairsCount: Int? = null
        private set

    var chairs: List<Chair>? = null
        set(chairs) {
            field = chairs
            this.chairsCount = chairs!!.size
        }

    constructor(cm: ConstructorMediator) : this(cm.id, cm.title) {
    }

    override fun toString(): String {
        return "Faculty{" +
                "id='" + id + '\'' +
                ", title='" + title + '\'' +
                '}'
    }
}
